main.go

```diff
@@ -31,11 +32,15 @@ func main() {
 	}
 	for _, table := range tables {
 		f.Table = table
-		fields, err := descriptor.Describe(db, table)
+		describeQuery := driver.Describe(f.Table)
```

Comment:
How about to merge these together into a driver.Fields which can do the Describe, Query and ParseFields? After that the interface has the Fields method in it.